### ISSUE TITLE
Pick how to handle conflicts in filesystem case sensitivity or insensitivity

### ISSUE THREAD

**Issue by ferd** [ferd/ReVault#8]
OSX is, by default, case insensitive.
So is Windows under NTFS and FAT32.
So is ExFAT.

Linux and BSD filesystems (ext3/4, UFS) are case sensitive.

ZFS is configurable.

A decision is required on how to handle these and make them work.

**Comment by llelf** [comment]
Also: Unicode normalization.
Examples:
- HFS+ normalizes (NFD) on write;
- APFS compares under normalization;
- ZFS configurable

**Comment by ferd** [comment]
yeah. I've just updated the model I'm using for tests to deal with case sensitivity and to use normalization there, it would be rather simpler to add a form of normalization there to force all tests to work with them.

So far I'm thinking of internally being case insensitive with ~250 chars limit (windows-like); If better solutions appear later, it's simpler to relax rules than constrain them. The big question is more about at which level should the detection of conflict be done (on-scan and on-sync, but lowest or higher processing layers?), and how to notify the user about it. So far there's nothing existing so there's no clear case of how to do things.
